Libraries/LibWeb/Crypto/CryptoAlgorithms.cpp

```diff
@@ -4049,8 +4049,29 @@ WebIDL::ExceptionOr<GC::Ref<CryptoKey>> ECDSA::import_key(AlgorithmParams const&
         if (!named_curve.is_empty() && named_curve != normalized_algorithm.named_curve)
             return WebIDL::DataError::create(m_realm, "Invalid algorithm"_string);
 
-        // TODO: 12. If the public key value is not a valid point on the Elliptic Curve identified
-        //           by the namedCurve member of normalizedAlgorithm throw a DataError.
+        // 12. If the public key value is not a valid point on the Elliptic Curve identified
+        //     by the namedCurve member of normalizedAlgorithm throw a DataError.
+        auto curve = [&named_curve] -> Variant<::Crypto::Curves::SECP256r1, ::Crypto::Curves::SECP384r1, ::Crypto::Curves::SECP521r1> {
```

Comment:
a follow-up to move this logic to some sort of file-static helper function might be a good refactor. as it is now, the call sites are all slightly different as far as what sorts of keys are seen as valid candidates.